### PR TITLE
Move building C++ source files down from the top CMakeLists.txt

### PR DESCRIPTION
**Pull request by momoko-h**
Currently the top level CMakeLists.txt does most of the work required to set up, build, and install the project, except for the file names. This makes the file quite large and unwieldy. 

This PR aims to move most of the commands needed to build the C++ sources to a new list file in the src directory. Building the launcher is mostly done in its own list file.

I hope this new organization makes it more obvious what each's file responsibility is and to find one's way around the build setup.

The changes of the #includes are necessary because the src directory itself is no longer passed as an include directory to the compiler. This removes the current illogical mix of mostly includes with only the file name of the header itself and the few includes with relative pathnames.

### CODE REVIEW

**Comment by momoko-h** [comment]
The MacOS runner was cancelled due to the brownout scheduled for today so I am going to retry that one tomorrow. Nevertheless I think this PR is ready for review.

**Comment by selaux** [comment]
Looks good to me so far.

File: CMakeLists.txt [hunk 1]
```diff
@@ -84,24 +84,25 @@ set(INSTALL_LIB_DIR lib CACHE PATH "Installation directory for libraries")
...
 option(WITH_RUST_BINARIES "Include rust binaries in build" ON)
-option(WITH_ASAN "Build with address sanitizer" OFF)
-option(ENABLE_PCH "Use precompiled standard library headers during compilation" OFF)
 # @see LOCAL_LUA_LIB in dependencies/lib-lua/CMakeLists.txt
 # @see LOCAL_SOL_LIB in dependencies/lib-sol2/CMakeLists.txt
 # @see LOCAL_MINIAUDIO_LIB in dependencies/lib-miniaudio/CMakeLists.txt
 set(WITH_CUSTOM_LOCALE "" CACHE STRING "Set a custom locale at the start, leave empty to disable")
+# Options specific to building C++ files can be found in the list file for
+# the src directory. Options specific to building the launcher application
+# arelocated in the list file for the launcher directory.
```
Comment (selaux) [inline CMakeLists.txt:98]: Nit: Missing space between `are located`.

**Comment by lynxlynxlynx** [comment]
LGTM too, thanks!

**Pull request merged**